client/my-sites/hosting/main.js

```diff
@@ -233,6 +238,7 @@ export const clickActivate = () =>
 export default connect(
 	( state ) => {
 		const siteId = getSelectedSiteId( state );
+		const site = getSelectedSite( state );
```

Comment:
Probably this can fix the issue.

```suggestion
		const site = getSite( state, siteId );
```

Reply:
I see in other places we also do:

```
const site = getSelectedSite( state );
const siteId = get( site, 'ID' );
```